src/java/htsjdk/samtools/util/SamLocusIterator.java

```diff
@@ -353,9 +395,17 @@ private void accumulateSamRecord(final SAMRecord rec) {
         final int alignmentEnd    = rec.getAlignmentEnd();
         final int alignmentLength = alignmentEnd - alignmentStart;
 
+        // get the offset for an insertion if we are tracking them
```

Comment:
Now that it's a little more complicated, could you please extract lines ~398-408 into a separate method called something like `ensureAccumulatorIsPopulated()`?